types/json-file-plus/index.d.ts

```diff
@@ -2,36 +2,118 @@
 // Project: https://github.com/ljharb/json-file-plus
 // Definitions by: Daniel Byrne <https://github.com/danwbyrne>
 //                 Jordan Harband <https://github.com/ljharb>
+//                 BendingBender <https://github.com/BendingBender>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
-declare function readJSON(filename: string): Promise<readJSON.JSONFile>;
+/**
+ * Read from and write to JSON files, without losing formatting, to minimize diffs.
+ *
+ * @example
+ * import jsonFile = require('json-file-plus');
+ * import * as path from 'node:path'
+ *
+ * const filename = path.join(process.cwd(), 'package.json');
+ * const callback = (err, result) => {
+ *     // your code here
+ * };
+ *
+ * jsonFile(filename, (err, file) => {
+ *     if (err) { return doSomethingWithError(err); }
+ *
+ *     file.data; // Direct access to the data from the file
+ *     file.format; // extracted formatting data. change at will.
+ *
+ *     file.get('version'); // get top-level keys
+ *     file.get('version', callback); // get top-level key
+ *     file.get(); // get entire data
+ *     file.get(callback); // get entire data
+ *
+ *     // pass any plain object into "set" to merge in a deep copy
+ *     // please note: references will be broken.
+ *     file.set({
+ *         foo: 'bar',
+ *         bar: {
+ *             baz: true
+ *         }
+ *     });
+ *
+ *     file.remove('description'); // remove a specific key-value pair
+ *     file.remove('description', callback); // remove a specific key-value pair
+ *
+ *     // change the filename if desired
+ *     file.filename = path.join(process.cwd(), 'new-package.json');
+ *
+ *     // Save the file, preserving formatting.
+ *     file.save(callback).then(function () {
+ *         console.log('success!');
+ *     }).catch(function (err) {
+ *         console.log('error!', err);
+ *     });
+ * });
+ */
+declare function readJSON(
+    filename: string,
+    callback?: (err: Error | null, file: readJSON.JSONFile) => void,
+): Promise<readJSON.JSONFile>;
 
 declare namespace readJSON {
-  function sync(filename: string): JSONFile;
+    /**
+     * Read from JSON files synchronously.
+     */
+    function sync(filename: string): JSONFile;
 
-  interface Format {
-    readonly indent: number | '\t';
-    readonly trailing: boolean;
-  }
+    interface Format {
+        indent: number | '\t' | ' ';
+        trailing: boolean;
```

Comment:
why are these no longer readonly?

Reply:
Because they really aren't. And the readme explicitly states: "extracted formatting data. change at will."

Reply:
Fair enough.